ckb-bin/src/subcommand/init.rs

```diff
@@ -8,14 +14,108 @@ use ckb_script::Runner;
 
 const SECP256K1_BLAKE160_SIGHASH_ALL_ARG_LEN: usize = 20 * 2 + 2; // 42 = 20 x 2 + prefix 0x
 
+fn check_db_compatibility(path: PathBuf, stdin: &Stdin, stdout: &mut StdoutLock) {
+    if path.exists() {
+        let config = DBConfig {
+            path: path.clone(),
+            ..Default::default()
+        };
+        if let Some(Error::DBError(err_msg)) = RocksDB::open_with_error(&config, 1).err() {
+            if err_msg.contains("the database version is not matched") {
+                write!(stdout, "Database is not incompatible, remove {:?}?", path).unwrap();
+                stdout.flush().unwrap();
+
+                let mut input = String::new();
+                let _ = stdin.read_line(&mut input);
+
+                if ["y", "Y"].contains(&input.trim()) {
+                    if let Some(e) = fs::remove_dir_all(path).err() {
+                        eprintln!("{}", e);
+                    }
+                }
+            }
+        }
+    }
+}
+
 pub fn init(args: InitArgs) -> Result<(), ExitCode> {
+    let mut args = args;
+
+    let stdout = stdout();
+    let mut stdout = stdout.lock();
+    let stdin = stdin();
+
     if args.list_chains {
         for spec in AVAILABLE_SPECS {
             println!("{}", spec);
         }
         return Ok(());
     }
 
+    let exported = Resource::exported_in(&args.root_dir);
+    if !args.force && exported {
+        eprintln!("Config files already exists, use --force to overwrite.");
+
+        if args.interactive {
+            write!(stdout, "Overwrite config file now?").unwrap();
```

Comment:
Please extract all stdin/stdout related code fragment into a helper `prompt` function: 

```
    let input: String = prompt("Overwrite config file now?");
    if !["y", "Y"].contains(&input) {
        return Err(ExitCode::Failure);
    }   
```

Reply:
sure :) that will be more concise

Reply:
@quake done, pls review again :)